### PR TITLE
Encryption: Refactor secrets.Service initialization

### PR DESCRIPTION
**Pull request by joanlopez**
**What this PR does / why we need it**:

Similar to https://github.com/grafana/grafana/pull/51014/files and based on the changes present there, the aim of this pull request is the refactor of the existing code to make next required changes easier.

The idea here is to move the responsibility of initializing (or not) the KMS providers to a single place (the `secrets.Service`) to make it easier to reason. So, later will be easier to initialize these providers when needed (not only when envelope encryption is enabled but also when the rollback migration is triggered).

Additionally, I've added a `sync.Once` to prevent providers from being initialized multiple times. 
I guess it should be safe anyway, but I added it to prevent spending time on that more once (easier to reason as well). 

**Which issue(s) this PR fixes**:

Partially contributes to https://github.com/grafana/grafana-enterprise/issues/3420

**Special notes for your reviewer**:

Note this pull request should not cause any behavior change (the KMS provider has been changed as such but the global behavior of both pieces should be the same (providers should be initialized only when EE is enabled).

This goes with https://github.com/grafana/grafana-enterprise/pull/3491.

Thanks!

### CODE REVIEW

**Comment by marefr** [comment]
Looks like you need to rebase. Quite hard to review otherwise

File: pkg/services/kmsproviders/osskmsproviders/osskmsproviders.go [hunk 1]
```diff
@@ -24,10 +24,6 @@ func ProvideService(enc encryption.Internal, settings setting.Provider, features
 }
 
 func (s Service) Provide() (map[secrets.ProviderID]secrets.Provider, error) {
-	if s.features.IsEnabled(featuremgmt.FlagDisableEnvelopeEncryption) {
```
Comment (joanlopez) [inline pkg/services/kmsproviders/osskmsproviders/osskmsproviders.go:27]: We delegate that behavior to the `Provide` caller in order to unify all the usages of `featuremgmt.FlagDisableEnvelopeEncryption`.

**Review by undef1nd — APPROVED** [review]
Looks good 👍

File: pkg/services/secrets/manager/manager.go [hunk 2]
```diff
@@ -34,60 +34,73 @@ type SecretsService struct {
...
+
+	s.log.Info("Envelope encryption state", "enabled", enabled, "current provider", currentProviderID)
+
 	s.registerUsageMetrics()
 
 	return s, nil
 }
 
+func (s *SecretsService) InitProviders() (err error) {
+	s.pOnce.Do(func() {
```
Comment (undef1nd) [inline pkg/services/secrets/manager/manager.go:97]: I don't have any opinion about the need of using Once here. Probably it does not hurt.

**Pull request merged**